rls/src/main/java/io/grpc/rls/internal/RlsProtoConverters.java

```diff
@@ -0,0 +1,196 @@
+    }
+
+    @Override
+    protected RouteLookupRequest doBackward(RlsProtoData.RouteLookupRequest routeLookupRequest) {
+      return
+          RouteLookupRequest.newBuilder()
+              .setServer(routeLookupRequest.getServer())
+              .setPath(routeLookupRequest.getPath())
+              .setTargetType(routeLookupRequest.getTargetType())
+              .putAllKeyMap(routeLookupRequest.getKeyMap())
+              .build();
+    }
+  }
+
+  /**
+   * RouteLookupResponseConverter converts between {@link RouteLookupResponse} and {@link
+   * RlsProtoData.RouteLookupResponse}.
+   */
+  public static final class RouteLookupResponseConverter
+      extends Converter<RouteLookupResponse, RlsProtoData.RouteLookupResponse> {
+
+    @Override
+    protected RlsProtoData.RouteLookupResponse doForward(RouteLookupResponse routeLookupResponse) {
+      return
+          new RlsProtoData.RouteLookupResponse(
+              routeLookupResponse.getTarget(),
+              routeLookupResponse.getHeaderData());
+    }
+
+    @Override
+    protected RouteLookupResponse doBackward(RlsProtoData.RouteLookupResponse routeLookupResponse) {
+      return RouteLookupResponse.newBuilder()
+          .setTarget(routeLookupResponse.getTarget())
+          .setHeaderData(routeLookupResponse.getHeaderData())
+          .build();
+    }
+  }
+
+  /**
+   * RouteLookupConfigConverter converts between json map to {@link RouteLookupConfig}.
+   */
+  public static final class RouteLookupConfigConverter
+      extends Converter<Map<String, ?>, RouteLookupConfig> {
+
+    @Override
+    protected RouteLookupConfig doForward(Map<String, ?> json) {
+      List<GrpcKeyBuilder> grpcKeyBuilders =
+          GrpcKeyBuilderConverter
+              .covertAll(JsonUtil.checkObjectList(JsonUtil.getList(json, "grpcKeyBuilders")));
+      checkState(!grpcKeyBuilders.isEmpty(), "At least one GrpcKeyBuilders are expected");
+      checkUniqueName(grpcKeyBuilders);
+      String lookupService = JsonUtil.getString(json, "lookupService");
+      long timeout =
+          TimeUnit.SECONDS.toMillis(JsonUtil.getNumberAsLong(json, "lookupServiceTimeout"));
+      Long maxAge =
+          convertTimeIfNotNull(
+              TimeUnit.SECONDS, TimeUnit.MILLISECONDS, JsonUtil.getNumberAsLong(json, "maxAge"));
+      Long staleAge =
+          convertTimeIfNotNull(
+              TimeUnit.SECONDS, TimeUnit.MILLISECONDS, JsonUtil.getNumberAsLong(json, "staleAge"));
+      long cacheSize = JsonUtil.getNumberAsLong(json, "cacheSizeBytes");
+      List<String> validTargets = JsonUtil.checkStringList(JsonUtil.getList(json, "validTargets"));
+      String defaultTarget = JsonUtil.getString(json, "defaultTarget");
+      RequestProcessingStrategy strategy =
+          RequestProcessingStrategy
+              .valueOf(JsonUtil.getString(json, "requestProcessingStrategy").toUpperCase());
+      return new RouteLookupConfig(
+          grpcKeyBuilders,
+          lookupService,
+          timeout,
```

Comment:
nit: Use `/*  argName = */` to let linter check if the args are correctly ordered.